crates/net/eth-wire/src/protocol.rs

```diff
@@ -52,11 +52,12 @@ impl Protocol {
     }
 
     /// The number of values needed to represent all message IDs of capability.
-    pub fn messages(&self) -> u8 {
+    pub fn messages(&self) -> Result<u8, ParseVersionError> {
```

Comment:
I think we can avoid the result and should just return `self.messages` all the time and remove the is_eth check and expect the self.messages is initialized correctly

Reply:
Is there a guarantee that `messages` used to construct `Protocol` is valid for the version specified in `Capability`?

https://github.com/paradigmxyz/reth/blob/f179b3a46e9a173ed011a856fe39dfd1da8a5c77/crates/net/eth-wire/src/protocol.rs#L22-L24

should we place the version check at construction instead?

Reply:
it is assumed